wildfly-feature-pack-maven-plugin/src/main/java/org/jboss/provisioning/wildfly/build/WfFeaturePackBuildMojo.java

```diff
@@ -328,6 +301,38 @@ private void doExecute() throws MojoExecutionException, MojoFailureException {
         }
     }
 
+    private void addModulesAll(final Path srcModulesDir, final FeaturePackLayout.Builder fpBuilder, final Path targetResources, final Path fpPackagesDir) throws MojoExecutionException {
+        getLog().info("WfFeaturePackBuildMojo adding modules.all");
+        final PackageSpec.Builder modulesAll = PackageSpec.builder(WfConstants.MODULES_ALL);
+        try {
+            final Map<String, Path> moduleXmlByPkgName = findModules(srcModulesDir);
+            if (moduleXmlByPkgName.isEmpty()) {
+                throw new MojoExecutionException("Modules not found in " + srcModulesDir);
+            }
+            packageModules(fpBuilder, targetResources, moduleXmlByPkgName, fpPackagesDir, modulesAll);
+        } catch (IOException e) {
+            throw new MojoExecutionException("Failed to process modules content", e);
+        }
+        if (!fpDependencies.isEmpty()) {
+            for (Map.Entry<String, FeaturePackLayout> fpDep : fpDependencies.entrySet()) {
+                final FeaturePackLayout fpDepLayout = fpDep.getValue();
+                if (fpDepLayout.hasPackage(WfConstants.MODULES_ALL)) {
+                    modulesAll.addPackageDep(fpDep.getKey(), WfConstants.MODULES_ALL);
+                }
+                if (fpDepLayout.hasPackage(WfConstants.DOCS)) {
```

Comment:
Docs should not depend on the presence of the modules.

Reply:
Sorry I'm missing something there

Reply:
You are calling this method only if the modules are found. If they aren't the docs from the deps won't be collected.